src/include/optimizer/rule_impls.h

```diff
@@ -41,17 +41,19 @@ class InnerJoinCommutativity : public Rule {
 /**
  * @brief (A join B) join C -> A join (B join C)
  */
-
-class InnerJoinAssociativity : public Rule {
+class JoinAssociativity : public Rule {
  public:
-  InnerJoinAssociativity();
+  JoinAssociativity();
 
   bool Check(std::shared_ptr<OperatorExpression> plan,
              OptimizeContext *context) const override;
 
   void Transform(std::shared_ptr<OperatorExpression> input,
                  std::vector<std::shared_ptr<OperatorExpression>> &transformed,
                  OptimizeContext *context) const override;
+ private:
+  bool StrongPredicate(std::shared_ptr<OperatorExpression> plan,
```

Comment:
There should be some documentation as to why a strong predicate check is needed for outer join associativity.

Reply:
@GustavoAngulo Where do you want them to add that?

Reply:
We don't really have a central documentation for the optimizer yet, so above the implementation of the `StrongPredicate` function should be good.

Reply:
Hey, can we add this in https://github.com/cmu-db/peloton/blob/master/src/include/optimizer/util.h? It makes more sense add it as a helper function instead of a member function of the rule.